MedallionShell.Tests/Streams/MergedLinesEnumerableTestBase.cs

```diff
@@ -97,46 +97,37 @@ void TestOneThrows(bool reverse)
         TestOneThrows(reverse: true);
     }
 
-    [Test, Timeout(10000)] // something's wrong if it's taking more than 10 seconds
-    public void FuzzTest()
+    [Test, Timeout(5_000)] // something's wrong if it's taking more than 5 seconds
+    public async Task FuzzTest()
     {
-        var pipe1 = new Pipe();
-        var pipe2 = new Pipe();
+        Pipe pipe1 = new(), pipe2 = new();
 
         var asyncEnumerable = this.Create(new StreamReader(pipe1.OutputStream), new StreamReader(pipe2.OutputStream));
 
         var strings1 = Enumerable.Range(0, 2000).Select(_ => Guid.NewGuid().ToString()).ToArray();
         var strings2 = Enumerable.Range(0, 2300).Select(_ => Guid.NewGuid().ToString()).ToArray();
 
-        static void WriteStrings(IReadOnlyList<string> strings, TextWriter writer)
+        static void WriteStrings(IReadOnlyList<string> strings, Pipe pipe)
         {
-            var spinWait = default(SpinWait);
-            var random = new Random(Guid.NewGuid().GetHashCode());
+            SpinWait spinWait = default;
+            Random random = new(Guid.NewGuid().GetHashCode());
+            using StreamWriter writer = new(pipe.InputStream);
             foreach (var line in strings)
             {
-                if (random.Next(110) == 1)
+                if (random.Next(10) == 1)
                 {
                     spinWait.SpinOnce();
                 }
 
                 writer.WriteLine(line);
             }
         }
-        
-        var task1 = Task.Run(() =>
-        {
-            using StreamWriter writer1 = new(pipe1.InputStream);
-            WriteStrings(strings1, writer1);
-        });
-        var task2 = Task.Run(() =>
-        {
-            using StreamWriter writer2 = new(pipe2.InputStream);
-            WriteStrings(strings2, writer2);
-        });
-        var consumeTask = asyncEnumerable.ToListAsync();
-        Task.WaitAll(task1, task2, consumeTask);
 
-        CollectionAssert.AreEquivalent(strings1.Concat(strings2).ToList(), consumeTask.Result);
+        var task1 = Task.Run(() => WriteStrings(strings1, pipe1));
+        var task2 = Task.Run(() => WriteStrings(strings2, pipe2));
+        Task.WaitAll(task1, task2); // need to dispose the writer to end the stream
```

Comment:
@Bartleby2718 can we try `await Task.WhenAll(task1, task2, consumeTask);` here?

Reply:
@madelson I've been trying to get that working, but this test fails for the async case if I do that. Not sure if the test logic is flawed (i.e. shouldn't await consumeTask if the input streams may not have been closed?) or there's a bug somewhere else.

Reply:
Did you try specifically with the test being async and using `await Task.WhenAll` instead of `Task.WaitAll`? If that doesn't work, could be some kind of threading bug in 1.7

Reply:
@madelson Yes, `MergedLinesEnumerableTestAsync`'s `FuzzTest` fails but `MergedLinesEnumerableTestSync`'s `FuzzTest` is fine. It fails even when it's run alone, but I did notice that `TestPipeline(2)` always fails with it if all tests are run together:
![image](https://github.com/madelson/MedallionShell/assets/7676354/94eb091b-caaf-4665-a41d-8f5c638e597a)

Reply:
@madelson I somehow got the test to pass with `await Task.WhenAll(task1, task2, consumeTask);`! Not sure if this is _the_ fix or it means something else needs to be fixed, but this does look promising.

Let me know what you think! (FWIW the test didn't pass within 10 secodns with `if (random.Next(4) == 1)`.)